share/functions/help.fish

```diff
@@ -23,58 +22,58 @@ function help --description 'Show help for the fish shell'
 	# by the help function defined below.
 	#
 	set -l graphical_browsers htmlview x-www-browser firefox galeon mozilla konqueror epiphany opera netscape rekonq google-chrome chromium-browser
-	set -l text_browsers htmlview www-browser links elinks lynx w3m
 
-	if type -q "$BROWSER"
-		# User has manually set a preferred browser, so we respect that
-		set fish_browser $BROWSER
-
-		# If browser is known to be graphical, put into background
-		if contains -- $BROWSER $graphical_browsers
-			set fish_browser_bg 1
-		end
+	if set -q fish_help_browser[1]
+		# User has set a fish-specific help browser. This overrides the
+		# browser that may be defined by $BROWSER. The fish_help_browser
+		# variable may be an array containing a browser name plus options.
+		set fish_browser $fish_help_browser
 	else
-		# Check for a text-based browser.
-		for i in $text_browsers
-			if type -q -f $i
-				set fish_browser $i
-				break
-			end
-		end
-
-		# If we are in a graphical environment, check if there is a graphical
-		# browser to use instead.
-		if test "$DISPLAY" -a \( "$XAUTHORITY" = "$HOME/.Xauthority" -o "$XAUTHORITY" = "" \)
-			for i in $graphical_browsers
+		set -l text_browsers htmlview www-browser links elinks lynx w3m
+
+		if set -q BROWSER
+			# User has manually set a preferred browser, so we respect that
+			set fish_browser $BROWSER
+		else
+			# Check for a text-based browser.
+			for i in $text_browsers
 				if type -q -f $i
 					set fish_browser $i
-					set fish_browser_bg 1
 					break
 				end
 			end
-		end
-
-		# If the OS appears to be Windows (graphical), try to use cygstart
-		if type -q cygstart
-			set fish_browser cygstart
-		# If xdg-open is available, just use that
-		else if type -q xdg-open
-			set fish_browser xdg-open
-		end
 	
+			# If we are in a graphical environment, check if there is a graphical
+			# browser to use instead.
+			if test "$DISPLAY" -a \( "$XAUTHORITY" = "$HOME/.Xauthority" -o "$XAUTHORITY" = "" \)
+				for i in $graphical_browsers
+					if type -q -f $i
+						set fish_browser $i
+						break
+					end
+				end
+			end
+
+			# If the OS appears to be Windows (graphical), try to use cygstart
+			if type -q cygstart
+				set fish_browser cygstart
+			# If xdg-open is available, just use that
+			else if type -q xdg-open
+				set fish_browser xdg-open
+			end
 	
-		# On OS X, we go through osascript by default
-		if test (uname) = Darwin
-			if type -q osascript
-				set fish_browser osascript
+			# On OS X, we go through osascript by default
+			if test (uname) = Darwin
+				if type -q osascript
+					set fish_browser osascript
+				end
 			end
 		end
-
 	end
 
-	if test -z $fish_browser
+	if set -q $fish_browser[1]
```

Comment:
That "$" needs to be removed, otherwise you're checking if the value of $fish_browser is the name of a set variable.

`set -q` takes just a variable name. Also `set -q var[1]`  is equivalent to `set -q var`.